### PR TITLE
[KOGITO-7412] - Generating operationId on OpenAPI spec descriptors for workflows rest interfaces

### PR DESCRIPTION
**Pull request by ricardozanini**
In this PR we introduce the Serverless Workflow `ConfigSource` to add a few default properties to the Quarkus add-ons that come by default with the workflow extensions.

See: https://issues.redhat.com/browse/KOGITO-7412

I'll make a few changes in the examples to remove the default properties and the smallrye-openapi dependency.

cc @krisv @pefernan 

Related PRs:

- https://github.com/kiegroup/kogito-examples/pull/1303

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

- <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/config/ServerlessWorkflowConfigSource.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    private static final Map<String, String> configuration = new HashMap<>();
+
+    /**
+     * Forcing the openapi spec generation of the workflow to add the operationId in the resulting file.
+     * 
+     * @see <a href="https://issues.redhat.com/browse/KOGITO-7412">Openapi files generated for ServerlessWorkflow doesn't contain operationId attribute</a>
+     * @see <a href="https://quarkus.io/guides/openapi-swaggerui#auto-generation-of-operation-id">Auto-generation of Operation Id</a>
+     */
+    private static final String OPENAPI_GEN_STRATEGY = "quarkus.smallrye-openapi.operation-id-strategy";
```
Comment (cristianonicolai) [inline quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/config/ServerlessWorkflowConfigSource.java:38]: @ricardozanini cant this be done by simply adding an application.properties in the extension with this value?

**Comment by cristianonicolai** [comment]
Jenkins run native

**Review by pefernan — APPROVED** [review]
+1 works fine! thanks @ricardozanini

**Review by krisv — CHANGES_REQUESTED** [review]
Great, hiding more config if not needed!  Yes, let's try to update the SW examples to remove properties if they are no longer needed.  This will make the examples easier and more in line with what they will see if they follow the steps themselves.

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/config/ServerlessWorkflowConfigSource.java (hunk 1 above)
Comment (ricardozanini) [reply quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/config/ServerlessWorkflowConfigSource.java:38]: no

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/config/ServerlessWorkflowConfigSource.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    private static final String DEFAULT_OPENAPI_GEN_STRATEGY = "METHOD";
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/config/ServerlessWorkflowConfigSource.java:39]: This is the default for any SW project. If the user wants to change it, they can just create an application.properties file or any other Quarkus config method to override it. I linked in the constant documentation a description of this behavior from Quarkus docs, see: https://quarkus.io/guides/openapi-swaggerui#auto-generation-of-operation-id

**Comment by ricardozanini** [comment]
> 

Kris, the examples are already updated and the PR is linked in the description.

CI error is not related. @cristianonicolai @jstastny-cz please take another look since this is quite urgent for 1.24.

**Review by krisv — APPROVED** [review]
Sorry, I missed that before, will take a look there.

**Review by jstastny-cz — APPROVED** [review]
Looks good, thanks.

**Pull request merged**